api/envoy/config/core/v3/protocol.proto

```diff
@@ -350,6 +350,11 @@ message Http1ProtocolOptions {
   // (inferred if not present), host (from the host/:authority header) and path
   // (from first line or :path header).
   bool send_fully_qualified_url = 8;
+
+  // If set, force HTTP/1 parser: BalsaParser if true, http-parser if false.
+  // If unset, HTTP/1 parser is selected based on
+  // envoy.reloadable_features.http1_use_balsa_parser.
+  optional bool use_balsa_parser = 9;
```

Comment:
I'd be inclined to have a sub-message for migration that's not-implemented-hide, so we can actually remove it after all the work

Reply:
Done.